### ISSUE TITLE
Announce selected or not while option is focused in the list menu of select box

### ISSUE THREAD

**Issue by mohit181191** [JedWatson/react-select#4343]
While navigating the options through the list menu using keyboard up/down arrow key. The screen reader announces the focused option and its position. Example below
If the option is X its position in 2 out 6
`option X focused, 2 of 6. 6 results available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu, press Tab to select the option and exit the menu.`

It does not announce if the option is already selected and exists in the selected list of options. The users who use screen readers would only be notified when it is actually selected or deselected using keyboard enter key.

Expected Behaviour(something similar to the below example)
`option X focused, 2 of 6. selected. 6 results available. Use Up and Down to choose options, press Enter to select the currently focused option, press Escape to exit the menu, press Tab to select the option and exit the menu.`

It would only be applicable when `hideSelectedOptions` prop is set to false

**Comment by ebonow** [comment]
Greetings @mohit181191,

I have addressed the work required for this as a part of PR #4414 which is being reviewed. This issue will be closed once it is merged.
